regex/src/main/scala/regex/Regex.scala

```diff
@@ -15,13 +29,51 @@ package regex
   */
 
 /** Simplifies a regular language */
-def simplify(lang: RegularLanguage): RegularLanguage = ???
+def simplify(lang: RegularLanguage): RegularLanguage = 
+  lang match
```

Comment:
I think it was a smart decision to have case statements over the key regular expression types. We didn't do that (basically just pattern-matched the given rules list verbatim), and it would've helped readability